[Technical-support] Prompt3-NullPointer: Add condition to avoid NullPointer when portlet is null

tototrinh opened the pull request:
I updated my work for this Prompt. Please help me review it. Thanks

holatuwol commented:
We see a lot of `Validator.isNull` calls for String objects, because there is a specific version of `isNull` for Strings.

https://github.com/tototrinh/liferay-portal/blob/debug-prompt-frontend-null-pointer/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L1065-L1106

However, there isn't one for Portlet, so it will end up using Object. We know that `portlet` is neither a Long nor a String, so the extra `instanceof` checks in the `isNull` implementation are wasteful when we can check for `null` directly.

https://github.com/tototrinh/liferay-portal/blob/debug-prompt-frontend-null-pointer/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L1040-L1053

Can you update your change to be a simpler null check?

tototrinh commented:
Hi @holatuwol
I have updated this PR. I have changed the way I check the portlet is null. Please review it again and leave your comments.
Thank you.

holatuwol commented:
This looks fine to me.

Please see [LPS-73444](https://issues.liferay.com/browse/LPS-73444) for additional context.
